cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/loadbalancer/AzureLoadBalancer.java

```diff
@@ -0,0 +1,71 @@
+package com.sequenceiq.cloudbreak.cloud.azure.loadbalancer;
+
+import com.sequenceiq.cloudbreak.cloud.model.CloudLoadBalancer;
+import com.sequenceiq.common.api.type.LoadBalancerType;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+
+public final class AzureLoadBalancer {
+    private static final String LOAD_BALANCER_NAME_PREFIX = "LoadBalancer";
+
+    private final List<AzureLoadBalancingRule> rules;
+
+    private final Set<AzureLoadBalancerProbe> probes;
+
+    private final String name;
+
+    private final LoadBalancerType type;
+
+    private final String instanceGroupName;
+
+    public AzureLoadBalancer(CloudLoadBalancer cloudLoadBalancer, String stackName) {
+        rules = cloudLoadBalancer.getPortToTargetGroupMapping()
+                .keySet()
+                .stream()
+                .map(AzureLoadBalancingRule::new)
+                .collect(toList());
+
+        // we want to derive the set of probes from the rules used in the load balancer
+        probes = rules.stream()
+                .map(AzureLoadBalancingRule::getProbe)
+                .collect(toSet());
+
+        this.name = getLoadBalancerName(cloudLoadBalancer.getType(), stackName);
+        this.type = cloudLoadBalancer.getType();
+        // we validate there's only one target instance group when the
+        // load balancer object is first initialized, so get the first
+        // and only element in the set
+        this.instanceGroupName = cloudLoadBalancer.getPortToTargetGroupMapping().values().stream()
+            .map(groups -> groups.iterator().next().getName())
+            .findFirst().get();
```

Comment:
I've iterated through this a few times and think the way to go is to just have a full `Set` of instance group names live in the `AzureLoadBalancer`.

It'll blow up if we try to provision an Azure environment that actually uses it at the moment though (because we're using `basic` instead of `standard` skus).

There's still a `Multimap`, but I wound up creating it with nested for loops, rather than a stream:

```java
ListMultimap<String, AzureLoadBalancer> mapping = MultimapBuilder.hashKeys().arrayListValues().build();
for (AzureLoadBalancer lb : azureLoadBalancers) {
    for (String group : lb.getInstanceGroupNames()) {
        mapping.put(group, lb);
    }
}
Map<String, Collection<AzureLoadBalancer>> map = mapping.asMap();
```
vs.
```java
ListMultimap<AzureLoadBalancer, String> mapping = azureLoadBalancers.stream()
        .collect(Multimaps.flatteningToMultimap(
                lb -> lb,
                lb -> lb.getInstanceGroupNames().stream(),
                MultimapBuilder.linkedHashKeys().arrayListValues()::build
        ));


Map<String, Collection<AzureLoadBalancer>> map = Multimaps.invertFrom(
        mapping,
        MultimapBuilder.hashKeys().arrayListValues().build())
        .asMap();
```

I couldn't figure out a way to get the `Multimaps.toMultimap` method to work directly on a `Set<String>` rather than `String` as the first argument.